File: examples/flask-realworld/app/profiles.py
from flask import Blueprint
from flask_jwt_extended import jwt_required, jwt_optional
from flask_apispec import marshal_with
from marshmallow import Schema, fields

profiles_bp = Blueprint('profiles', __name__)

class ProfileSchema(Schema):
    username = fields.String()
    bio = fields.String()
    image = fields.Url()
    following = fields.Boolean()

@profiles_bp.route('/<username>', methods=['GET'])
@jwt_optional
@marshal_with(ProfileSchema)
def get_profile(username):
    """Get a user's profile."""
    pass

@profiles_bp.route('/<username>/follow', methods=['POST'])
@jwt_required
def follow_user(username):
    """Follow a user."""
    pass

@profiles_bp.route('/<username>/follow', methods=['DELETE'])
@jwt_required
def unfollow_user(username):
    """Unfollow a user."""
    pass

File: examples/vue-realworld/src/store/modules/profiles.js
import ApiService from '../../services/api.service';

const FETCH_PROFILE = 'fetchProfile';
const FOLLOW_USER = 'followUser';
const UNFOLLOW_USER = 'unfollowUser';

const state = {
  profile: null,
};

const actions = {
  [FETCH_PROFILE](context, username) {
    return ApiService.get(`profiles/${username}`);
  },
  [FOLLOW_USER](context, username) {
    return ApiService.post(`profiles/${username}/follow`);
  },
  [UNFOLLOW_USER](context, username) {
    return ApiService.delete(`profiles/${username}/follow`);
  },
};

export default { state, actions };
